packages/squiggle-lang/src/js/math.ts

```diff
@@ -3,19 +3,34 @@ type SampleArgs = {
   size: number;
 };
 
+/** Searches for a first element in `arr` greater than `value` */
+function binsearchFirstGreater(arr: number[], value: number): number {
+  let s = 0
+  let e = arr.length - 1;
+  while ((e - s) > 1) {
+    const i = Math.ceil((s + e) / 2)
+    if (arr[i] < value) {
+      s = i + 1
+    } else {
+      e = i
+    }
+  }
+  return value < arr[s] ? s : e
```

Comment:
Not a fan of this as it accesses past the end for a length-0 `arr` (true, it's not valid to sample from 0 possibilities). I think it can be removed if you use floor instead of ceil above and lower the termination condition to 0.